### PR TITLE
DMU 10 cards

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/warhosts_frenzy.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Name:Warhost's Frenzy
+ManaCost:2 R
+Types:Instant
+K:Kicker:B
+A:SP$ PumpAll | ValidCards$ Creature.YouCtrl | NumAtt$ 2 | SubAbility$ DBEffect | SpellDescription$ Creatures you control get +2/+0 until end of turn. If this spell was kicked, whenever a creature you control dies this turn, draw a card.
+SVar:DBEffect:DB$ Effect | Triggers$ TrigDies | Condition$ Kicked
+SVar:TrigDies:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Graveyard | ValidCard$ Creature | Execute$ TrigDraw | TriggerDescription$ Whenever a creature dies this turn, you gain 1 life.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/warhosts_frenzy.txt:7]: you control part missing

File: forge-gui/res/cardsfolder/upcoming/keldon_strike_team.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+ManaCost:2 R
+Types:Creature Human Soldiers
+PT:3/1
+K:Kicker:1 W
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self+kicked | Execute$ TrigToken | TriggerDescription$ When CARDNAME enters the battlefield, if it was kicked, create two 1/1 white Soldier creature tokens.
+SVar:TrigToken:DB$ Token | TokenAmount$ 2 | TokenScript$ w_1_1_soldier | TokenOwner$ You
+S:Mode$ Continuous | IsPresent$ Card.Self+ThisTurnEntered | Affected$ Creature.YouCtrl | AddKeyword$ Haste | Description$ As long as CARDNAME entered the battlefield this turn, creatures you control have haste.
+DeckHints:Color$White
+DeckHas:Ability$Token
+Oracle:Kicker {1}{W} (You may pay an additional {1}{W}} as you cast this spell.)\nWhen Keldon Strike Team enters the battlefield, if it was kicked, create two 1/1 white Soldier creature tokens.\nAs long as Keldon Strike Team entered the battlefield this turn, creatures you control have haste.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/keldon_strike_team.txt:11]: `{W}}` > `{W}`

File: forge-gui/res/cardsfolder/upcoming/keldon_strike_team.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Name:Keldon Strike Team
+ManaCost:2 R
+Types:Creature Human Soldiers
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/keldon_strike_team.txt:3]: `Soldiers` > `Soldier`

File: forge-gui/res/cardsfolder/upcoming/molten_monstrosity.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Name:Molten Monstrosity
+ManaCost:7 R
+Types:Creature Hellion
+PT:5/5
+K:Trample
+S:Mode$ ReduceCost | ValidCard$ Card.Self | Type$ Spell | Amount$ X | EffectZone$ All | Description$ CARDNAME costs {X} less to cast, where X is the greatest power among creatures you control.
+SVar:X:Count$GreatestPower_Creature.YouCtrl
```
[the remark this answers is not in the document]
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/molten_monstrosity.txt:7]: @tool4ever I think it was discerning_taste, not sure though but will change it

File: forge-gui/res/cardsfolder/upcoming/knight_of_dusks_shadow.txt [hunk 5]
```diff
@@ -0,0 +1,8 @@
+Name:Knight of Dusk's Shadow
+ManaCost:1 B
+Types:Creature Human Knight
+PT:2/2
+K:Menace
+S:Mode$ CantGainLife | ValidPlayer$ Player.Opponent | Description$ Your opponents can't gain life.
+A:AB$ Pump | Cost$ 1 B | Defined$ Self | NumAtt$ +1 | NumDef$ +1 | SpellDescription$ CARDNAME gets +1/+1 until end of turn.
+Oracle:Menace (This creature can't be blocked except by two or more creatures.)\nYour opponents can’t gain life.\n{1}{B}: Knight of Dusk's Shadow gets +1/+1 until end of turn.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/knight_of_dusks_shadow.txt:8]: `’` > `'`

File: forge-gui/res/cardsfolder/upcoming/micromancer.txt [hunk 6]
```diff
@@ -0,0 +1,9 @@
+Name:Micromancer
+ManaCost:3 U
+Types:Creature Human Wizard
+PT:3/3
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigChange | OptionalDecider$ You | TriggerDescription$ When CARDNAME enters the battlefield,you may search your library for an instant or sorcery card with mana value 1, reveal it, put it into your hand, then shuffle.
+SVar:TrigChange:DB$ ChangeZone | Origin$ Library | Destination$ Hand | ChangeType$ Instant.cmcEQ1,Sorcery.cmcEQ1 | ChangeNum$ 1
+SVar:BuffedBy:Instant,Sorcery
```
[the remark this answers is not in the document]
Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/micromancer.txt]: 49 instances exist – might be a fun cleanup 😉

File: forge-gui/res/cardsfolder/upcoming/micromancer.txt [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+DeckHints:Type$Instant|Sorcery
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/micromancer.txt:7]: Change this to `Needs` I think

File: forge-gui/res/cardsfolder/upcoming/micromancer.txt [hunk 8]
```diff
@@ -0,0 +1,9 @@
+Name:Micromancer
+ManaCost:3 U
+Types:Creature Human Wizard
+PT:3/3
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigChange | OptionalDecider$ You | TriggerDescription$ When CARDNAME enters the battlefield,you may search your library for an instant or sorcery card with mana value 1, reveal it, put it into your hand, then shuffle.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/micromancer.txt:5]: `battlefield,you` > `battlefield, you`

File: forge-gui/res/cardsfolder/upcoming/monstrous_war_leech.txt [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+ManaCost:3 B
+Types:Creature Leech Horror
+PT:*/*
+K:Kicker:U
+K:ETBReplacement:Other:TrigMill
+SVar:TrigMill:DB$ Mill | NumCards$ 4 | Defined$ You | Condition$ Kicked
+S:Mode$ Continuous | EffectZone$ All | CharacteristicDefining$ True | SetPower$ X | SetToughness$ X | Description$ CARDNAME's power and toughness are each equal to the highest mana value among cards in your graveyard.
+SVar:X:Count$HighestCMC_Creature.YouCtrl+inZoneGraveyard
+DeckHas:Ability$Graveyard|Mill
+SVar:NoZeroToughnessAI:True
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/monstrous_war_leech.txt:11]: add `DeckHints:Color$Blue`

File: forge-gui/res/cardsfolder/upcoming/monstrous_war_leech.txt [hunk 10]
```diff
@@ -0,0 +1,12 @@
+Name:Monstrous War-Leech
+ManaCost:3 B
+Types:Creature Leech Horror
+PT:*/*
+K:Kicker:U
+K:ETBReplacement:Other:TrigMill
+SVar:TrigMill:DB$ Mill | NumCards$ 4 | Defined$ You | Condition$ Kicked
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/monstrous_war_leech.txt:7]: Needs SpellDescription$

File: forge-gui/res/cardsfolder/upcoming/molten_monstrosity.txt [hunk 11]
```diff
@@ -0,0 +1,8 @@
+Name:Molten Monstrosity
+ManaCost:7 R
+Types:Creature Hellion
+PT:5/5
+K:Trample
+S:Mode$ ReduceCost | ValidCard$ Card.Self | Type$ Spell | Amount$ X | EffectZone$ All | Description$ CARDNAME costs {X} less to cast, where X is the greatest power among creatures you control.
+Count$Valid Creature.YouCtrl$GreatestPower
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/molten_monstrosity.txt:7]: SVar:X: still need ofc

File: forge-gui/res/cardsfolder/upcoming/monstrous_war_leech.txt [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+S:Mode$ Continuous | EffectZone$ All | CharacteristicDefining$ True | SetPower$ X | SetToughness$ X | Description$ CARDNAME's power and toughness are each equal to the highest mana value among cards in your graveyard.
+SVar:X:Count$HighestCMC_Creature.YouCtrl+inZoneGraveyard
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/monstrous_war_leech.txt:9]: I think `Ctrl` > `Own`  ??

File: forge-gui/res/cardsfolder/upcoming/micromancer.txt [hunk 13]
```diff
@@ -0,0 +1,8 @@
+Name:Micromancer
+ManaCost:3 U
+Types:Creature Human Wizard
+PT:3/3
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigChange | OptionalDecider$ You | TriggerDescription$ When CARDNAME enters the battlefield, you may search your library for an instant or sorcery card with mana value 1, reveal it, put it into your hand, then shuffle.
+SVar:TrigChange:DB$ ChangeZone | Origin$ Library | Destination$ Hand | ChangeType$ Instant.cmcEQ1,Sorcery.cmcEQ1 | ChangeNum$ 1
+DeckNeeds:Type$Instant|Sorcery
+Oracle:When Micromancer enters the battlefield, you may search your library for an instant or sorcery card with mana value 1, reveal it, put it into your hand, then shuffle.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/micromancer.txt:8]: Hmm, is the cmcEQ1 enough to warrant `AI:RemoveDeck:Random` ? It's a pretty narrow condition for this card to be useful...

**Review by Northmoc — APPROVED** [review]
Looks good to me!

File: forge-gui/res/cardsfolder/upcoming/micromancer.txt (hunk 6 above)
[the remark this answers is not in the document]
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/micromancer.txt]: I checked and I was wrong actually

**Pull request merged**